### PR TITLE
Simplified plugin config check + apply_os_env

### PR DESCRIPTION
**Pull request by uwiger**
See issue #3303 
This PR simplifies the config checking for the plugin, and allows the plugin to define an OS env prefix, similar to the `AE` prefix used by the main node. This will allow for parameterization using OS env variables like `DEV__KEYBLOCK_INTERVAL=10`.

### CODE REVIEW

**Review by velzevur — DISMISSED** [review]

**Comment by marc0olo** [comment]
I don't feel confident in reviewing Erlang code. will leave that up to others :)

File: apps/aeutils/src/aeu_env.erl [hunk 1]
```diff
@@ -305,23 +310,34 @@ read_config(Mode) when Mode =:= silent; Mode =:= report ->
 
 apply_os_env() ->
     ok = application:ensure_started(gproc),
-    try
-    %% Make sure gproc is started, since we use it in update_config/1
-    application:ensure_started(gproc),
     Pfx = "AE",  %% TODO: make configurable
```
Comment (nikita-fuchs) [inline apps/aeutils/src/aeu_env.erl:313]: shouldn't this be `DEV` or so then ?

File: apps/aeutils/src/aeu_env.erl [hunk 2]
```diff
@@ -559,17 +582,30 @@ lst(L) when is_list(L) -> L;
 lst(E) -> [E].
 
 update_config(Map) when is_map(Map) ->
+    update_config(Map, _Notify = true).
```
Comment (nikita-fuchs) [inline apps/aeutils/src/aeu_env.erl:585]: does this generally mean we can update some or all configs also in runtime ?

Comment (uwiger) [reply apps/aeutils/src/aeu_env.erl:585]: Generally, it sorta does, but also generally, not all code in the system may be prepared for runtime changes of the user config, so your mileage may vary at this time.

File: apps/aeutils/src/aeu_env.erl (hunk 1 above)
Comment (uwiger) [reply apps/aeutils/src/aeu_env.erl:313]: This function is called during the normal boot. The support function, called from `aeu_plugins` is flexible in regard to prefix and schema. The thing that is not yet configurable is the os env prefix for overriding the parameters in the main schema.

**Review by velzevur — APPROVED** [review]

**Review by velzevur — APPROVED** [review]

**Comment by marc0olo** [comment]
any chance to get other reviews here? again, I am probably not the right guy to review Erlang code :D

**Comment by hanssv** [comment]
Ulf promises to fix everything later, so I'm good!

**Review by hanssv — APPROVED** [review]

**Pull request merged**